tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc

```diff
@@ -126,6 +127,29 @@ __global__ void UnsortedSegmentCustomKernel(
   }
 }
 
+bool RequireDeterminism() {
+  static bool require_determinism = [] {
+    bool deterministic_ops = false;
+    TF_CHECK_OK(tensorflow::ReadBoolFromEnvVar("TF_DETERMINISTIC_OPS",
+                                               /*default_val=*/false,
+                                               &deterministic_ops));
+    return deterministic_ops;
+  }();
+  return require_determinism;
+}
```

Comment:
Should this live in a single utility somewhere?

Reply:
Probably. I've tried moving it up into a utility in previous PRs but I believe it wasn't accepted because we didn't want to institutionalize something that was not intended to the long-term correct way of doing it (i.e. we'll eventually move to config). I can (re-)scout for a location, move it up, and share it with other code.

Reply:
I added a TODO comment to this. How about I do this (lower-priority) factoring later and in a separate PR? It's going to potentially have a wide span in the TensorFlow source code. This function is currently replicated in several places.